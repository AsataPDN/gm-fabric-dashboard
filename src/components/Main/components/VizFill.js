import styled from "styled-components";
import { PropTypes } from "prop-types";

import { contrastColor } from "style/styleFunctions";

const VizFill = styled.div`
  width: ${props => (props.width ? `${props.width}%` : "0%")};
  position: absolute;
  background-color: ${props =>
    props.width
      ? `${contrastColor("red", props.width / 100, "blue")}`
      : "currentColor"};
  border-radius: 0 1px 0 0;
  height: 2px;
  left: 0;
  bottom: 0;
`;

VizFill.propTypes = {
  width: PropTypes.number
};

export default VizFill;
