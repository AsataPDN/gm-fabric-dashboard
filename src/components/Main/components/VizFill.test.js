import React from "react";
import { shallow } from "enzyme";
import VizFill from "./VizFill";

describe("VizFill", () => {
  it("matches snapshot", () => {
    const aVizFill = shallow(<VizFill />);
    expect(aVizFill).toMatchSnapshot();
  });
});
