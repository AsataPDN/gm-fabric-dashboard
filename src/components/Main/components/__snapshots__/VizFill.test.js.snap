// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VizFill matches snapshot 1`] = `
.c0 {
  background-color: currentColor;
  border-radius: inherit;
  bottom: 0;
  color: currentColor;
  left: 0;
  min-width: 2px;
  position: absolute;
  top: 0;
  width: 0%;
}

<div
  className="c0"
/>
`;
